lib/puppet/provider/package/pip3.rb

```diff
@@ -0,0 +1,36 @@
+# Puppet package provider for Python's `pip3` package management frontend.
+# <http://pip.openplans.org/>
+
+require 'puppet/provider/package/pip'
+
+Puppet::Type.type(:package).provide :pip3,
+  :parent => :pip do
+
+  desc "Python packages via `pip3`.
+
+  This provider supports the `install_options` attribute, which allows command-line flags to be passed to pip3.
+  These options should be specified as a string (e.g. '--flag'), a hash (e.g. {'--flag' => 'value'}),
+  or an array where each element is either a string or a hash."
+
+  def self.cmd
+    if Facter.value(:osfamily) == "RedHat" and Facter.value(:operatingsystemmajrelease).to_i < 7
+      "pip-python3"
+    else
+      "pip3"
+    end
+  end
+
+  # Execute a `pip3` command.  If Puppet doesn't yet know how to do so,
+  # try to teach it and if even that fails, raise the error.
+  private
+  def lazy_pip(*args)
+    pip3 *args
+  rescue NoMethodError => e
+    if pathname = which(self.class.cmd)
+      self.class.commands :pip3 => pathname
+      pip3 *args
```

Comment:
It looks like the original `lazy_pip` is already using `self.class.cmd`, so I think it already should just work correctly without a duplicated method.